standard/interfaces.md

```diff
@@ -230,92 +230,92 @@ interface_member_declaration
     ;
 ```
 
-An interface declaration declares zero or more members. The members of an interface shall be constants, fields, methods, properties, events, indexers, operators, constructors, and types, some of which may be instance, others static, as described in the subclauses for each interface member kind. Interfaces shall not declare instance data members, including field like events.
+This clause augments the description of members in classes (§15.3) with restrictions for interfaces. The Interface members are declared using *member_declaration*s with the following additional rules:
 
-All interface members implicitly have public access; however, an explicit access modifier ([§7.5.2](basic-concepts.md#752-declared-accessibility)) is permitted except on static constructors (§15.12).
+- A *finalizer_declaration* is not allowed.
+- Instance constructors, *constructor_declaration*s, are not allowed.
+- All interface members implicitly have public access; however, an explicit access modifier ([§7.5.2](basic-concepts.md#752-declared-accessibility)) is permitted except on static constructors (§15.12).
+- An interface instance function member whose declaration includes a body is an implicitly `virtual` member unless the `sealed` or `private` modifier is used.
+- The `virtual` modifier may be used on a function member that would otherwise be implicitly `virtual`.
+- The `abstract` modifier is implied for interface function members without bodies; that modifier may be given explicitly.
+- A derived interface may override an abstract member declared in a base interface. A non-virtual instance function member may be declared using the `sealed` keyword.
+- A `private` or `sealed` function member of an interface shall have a body.
+- A `private` function member shall not have the modifier `sealed`.
+- An explicitly implemented function member shall not have the modifier `sealed`.
+- The name of a non-abstract interface member is only visible in that interface and interfaces derived from it. That member can be accessed only through a reference whose compile-time type is an interface type that is implicitly convertible to the interface where the non-abstract member is declared.
```

Comment:
(And, similar question about whether the term 'virtual' is in the sense of 'virtual call' and thus includes abstract members)
```suggestion
- The name of a non-virtual interface member is only visible in that interface and interfaces derived from it. That member can be accessed only through a reference whose compile-time type is an interface type that is implicitly convertible to the interface where the non-abstract member is declared.
```

Reply:
I'd like to remove this condition (including where this statement is mirrored ten lines down in the note). It is not due to the abstractness or non-abstractness that the name is visible within the class. Rather, it is due to whether the class declares a matching _class_ member.

```suggestion
- The name of an interface member is only visible in that interface and interfaces derived from it. That member can be accessed only through a reference whose compile-time type is an interface type that is implicitly convertible to the interface where the non-abstract member is declared.
```

Reply:
Reworked these rules in this section.